apps/chat/src/pages/api/publication/resourceListing.ts

```diff
@@ -0,0 +1,63 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import { getServerSession } from 'next-auth';
+import { getToken } from 'next-auth/jwt';
+
+import { validateServerSession } from '@/src/utils/auth/session';
+import { getApiHeaders } from '@/src/utils/server/get-headers';
+import { logger } from '@/src/utils/server/logger';
+
+import { DialAIError } from '@/src/types/error';
+
+import { errorsMessages } from '@/src/constants/errors';
+
+import { authOptions } from '@/src/pages/api/auth/[...nextauth]';
+
+import fetch from 'node-fetch';
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const session = await getServerSession(req, res, authOptions);
+  const isSessionValid = validateServerSession(session, req, res);
+  const token = await getToken({ req });
+  if (!isSessionValid) {
+    return;
+  }
```

Comment:
as I can see we shouldn't read token before session checking here in other handlers
```suggestion
  if (!isSessionValid) {
    return;
  }
  const token = await getToken({ req });
```

Reply:
I fixed it everywhere